### DISCUSSION TITLE
Auto-clean stored json files created by `app.storage`

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#991] [Ideas / Feature Requests]
Currently, stored json files are not cleaned up. Maybe this feature should be combined with the ability to set the `max_age` from the internally used SessionMiddelware...

**Comment by pythonormen** [comment]
Any plans on implementing this? Thanks

**Comment by rodja** [comment]
Yes, this should be implemented. Anyone interested in taking a stab? See [CONTRIBUTING.md](https://github.com/zauberzeug/nicegui/blob/main/CONTRIBUTING.md#pull-requests) for a short explanation on how to create pull requests.

**Reply by falkoschindler** [reply]
What would be the requirement? When exactly would a file be deleted? And what would the API be like?

**Reply by rodja** [reply]
For `storage.user` I think we can remove the files when the session cookie expires. `storage.browser` has no files. I think it is ok for `storage.general` to stay alive for ever. Cleaning up old entries would be in the responsibility of the app developers.

**Comment by rodja** [comment]
In https://github.com/zauberzeug/nicegui/discussions/1448 we have a request to change the max_age and other parameters of the `SessionMiddleware`. It should be tackled together with this.

**Reply by CrystalWindSnake** [reply]
Is this problem currently solved? Sometimes it is desirable to use a temporary session, for example by setting max_age to none.

**Reply by rodja** [reply]
Please keep the discussions separated and post your example over at #1448.

**Comment by rodja** [comment]
Somehow related is the discussion on how and when to clean up in-memory data in #2837 which introduces `app.storage.tab` as a place to store data across reloads within a specific tab.
